scripts/validate-pull-request.mjs

```diff
@@ -0,0 +1,69 @@
+// @ts-check
+
+/**
+ * A pull request is valid if:
+ * - it is a release PR (i.e. it's head ref starts with "changeset-release/")
+ * - or it is labeled as "no changeset needed"
+ * - or it has a changeset.
+ */
+
+import { exec as execSync } from "node:child_process";
+import { promisify } from "node:util";
+
+const exec = promisify(execSync);
+
+const changesetDir = ".changeset";
+
+function isReleasePR() {
+  if (process.env.GITHUB_HEAD_REF === undefined) {
+    throw new Error("GITHUB_HEAD_REF is not defined");
+  }
+
+  return process.env.GITHUB_HEAD_REF.startsWith("changeset-release/");
+}
+
+function hasNoChangesetNeededLabel() {
+  if (process.env.GITHUB_EVENT_PULL_REQUEST_LABELS === undefined) {
+    throw new Error("GITHUB_EVENT_PULL_REQUEST_LABELS is not defined");
+  }
+
+  const labels = JSON.parse(process.env.GITHUB_EVENT_PULL_REQUEST_LABELS);
+
+  return labels.some(l => l.name === "no changeset needed");
+}
+
+async function hasChangeset() {
+  if (process.env.GITHUB_BASE_REF === undefined) {
+    throw new Error("GITHUB_BASE_REF is not defined");
+  }
+
+  const { stdout } = await exec(
+    `git diff --name-only --diff-filter=A ${process.env.GITHUB_BASE_REF} -- ${changesetDir}`
+  );
+
+  const changesets = stdout.trim().split("\n")
+    .filter((file) => file.endsWith(".md"));
+
+  return changesets.length > 0;
+}
+
+async function validatePullRequest() {
```

Comment:
This is a nit pick, don't change the PR. In this sort of case I would put the `validatePullRequest` function at the top, to ease reading from top to bottom - while keeping the invocation at the very bottom.